x-pack/platform/plugins/shared/security/server/session_management/session_index.ts

```diff
@@ -629,10 +629,23 @@ export class SessionIndex {
     // Prior to https://github.com/elastic/kibana/pull/134900, sessions would be written directly against the session index.
     // Now, we write sessions against a new session index alias. This call ensures that the alias exists, and is attached to the index.
     // This operation is safe to repeat, even if the alias already exists. This seems safer than retrieving the index details, and inspecting
-    // it to see if the alias already exists.
+    // it to see if the alias already exists. If the index has been upgraded using the UA, then the index name has changed and `.kibana_security_session_1`
+    // is now an alias to the upgraded index. We must attach the alias to the index by using the current index name.
     try {
+      const aliasResponse = await this.options.elasticsearchClient.indices.getAlias({
+        name: this.aliasName,
+      });
+      const indexNames = Object.keys(aliasResponse);
+      const indexName = indexNames.length > 0 ? indexNames[0] : this.indexName;
+
+      if (!indexName) {
+        throw new Error(
```

Comment:
nit/observation: throwing/catching locally usually indicates that we are using exceptions improperly. However, I see a lot of the control flow of this class is based on this function throwing an exception and throwing before the 2nd call to ES is more efficient. No action needs to be taken here.